docs/DeveloperGuide.md

```diff
@@ -59,17 +60,29 @@ Checklist before raising an issue:
 liberally to assist in readability.
   * [Code fences](https://help.github.com/articles/creating-and-highlighting-code-blocks/) for exception stack traces and log entries, for example, massively improve readability.
 
-## Building and Testing
+## Building
 
 DataHelix uses Java 1.8 which can be downloaded from this [link](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html)
 
 DataHelix uses [gradle](https://gradle.org/) to automate the build and test process. To build the project run `gradle build` from the root folder of the project. If it was successful then the created jar file can be found in the path _orchestrator/build/libs/generator.jar_ .
 
 [Guice](https://github.com/google/guice) is used in DataHelix for Dependency Injection (DI). It is configured in the 'module' classes, which all extend `AbstractModule`, and injected with the `@inject` annotation.
 
+## Testing
+
+Our strategy is to ensure all aspects of the generator are tested through some form of automation testing as we strive to ensure correctness, quality and prevent against regression issues.
+
+* For all new classes and methods that are developed, unit or component tests should be added to the code base
+
+* If a change is made to existing class implementations and a test does not exist, then a test should be added
+
+* If it's a brand new feature, a Schema validation test should be added as well as an appropriate Cucumber test
```

Comment:
Might be good to explain what is meant by a "Schema Validation" test here, or to point to existing examples.

Reply:
Thanks Matt - this link has now been added.